task05/controllers/userController.js

```diff
@@ -0,0 +1,135 @@
+const {
+  getAll,
+  getById,
+  create,
+  update,
+  remove,
+} = require('../models/userModel');
+const { parseRequestBody, createUserResponse } = require('../utils/utils');
+
+async function getUsers(req, res) {
+  try {
+    const users = await getAll();
+    res.writeHead(200, {
+      'Content-Type': 'application/json',
+      'Cache-Control': 'public, max-age=3600',
+    });
+
+    res.end(
+      JSON.stringify({
+        data: users.map((user) => createUserResponse(user)),
+        error: null,
+      })
+    );
+  } catch (error) {
+    console.log(error);
+  }
+}
+
+async function getUserHobbies(req, res, id) {
+  try {
+    const user = await getById(id);
+    if (!user) {
+      res.writeHead(404, {
+        'Content-Type': 'application/json',
+        'Cache-Control': 'private, max-age=3600',
+      });
+      res.end(
+        JSON.stringify({
+          data: null,
+          error: `User with id ${id} doesn't exist`,
+        })
+      );
```

Comment:
Considering that this logic is used everywhere in the controller, it can be moved out for reuse.